### ISSUE TITLE
Cursor routing don't work with IntelliJ idea community edition 2019.1.1, NVDA 2019.1.1 and HumanWare Brailliant BI 40

### ISSUE THREAD

**Issue by Nitram1123** [nvaccess/nvda#9512]
### Steps to reproduce:
Run NVDA
Run IntelliJ
Open a java file in the editor
Test the cursor routing

### Expected behavior:
No button works on my Brailliant BI in the editor

#### NVDA version:
2019.1.1

#### Windows version:
Windows 10

#### Name and version of other software in use when reproducing the issue:
Java jre 1.8.0_172

### Other questions
I believe this feature is supported by NVDA? If not, do you know when it will be?
[log-nvda.txt](https://github.com/nvaccess/nvda/files/3103376/log-nvda.txt)

**Comment by clementb49** [comment]
Hi
I think the problem is not caused by NVDA because the routing cursor doesn't work in IntelliJ based IDE with other screen readers like JAWS. 
For a Java Access Bridge Application, how does NVDA send that the cursor needs to be moved?

**Comment by mwhapples** [comment]
I agree that this looks like an IntelliJ issue. I have inserted additional logging statements into NVDA and that shows correct calculation of new cursor position based on routing keys and calling of Java Access Bridge setCursorPosition function with this value.
I also have launched the JDK demo notepad application and cursor routing works there.
Inspecting the accessibility information with access bridge explorer, I see one main difference between the IntelliJ editor and the demo application's text control. The demo application's text control has actions where as the IntelliJ editor exposes no actions. May this be linked to the issue?

**Comment by seanbudd** [comment]
Has this been reported to JetBins / IntelliJ?

**Comment by mwhapples** [comment]
I plan to report this to jetbrains today.

**Comment by mwhapples** [comment]
Here is a link to the JetBrains bug report I have created https://youtrack.jetbrains.com/issue/IDEA-299407/The-main-source-code-editor-view-in-IntelliJ-IDEA-and-other-jetBrains-IDEs-do-not-respond-to-accessibility-setCaretPosition

**Comment by mwhapples** [comment]
OK, my issue was closed and marked as duplicate of another issue, here is a link to that issue https://youtrack.jetbrains.com/issue/JBR-4599

**Comment by fcnjd** [comment]
Yes, last year I filled an automated IntelliJ survey where I wrote that there are a lot of accessibility issues, and after that I got contacted by technical support and they created the issues for me, including this one. I hope they'll improve that soon, as there are already a lot of users who wrote that they would find working routing keys helpful as well.

**Comment by Adriani90** [comment]
How is this now behaving in NVDA 2024.2 with IntelliJ Idea newer community versions? The last one is 2024.1.4.

**Comment by Adriani90** [comment]
@hamidrezaabroshan do you by chance have a braille display you could test with whether this issue still occurs in IntelliJ 2024.1.4?
cc: @burmancomp, @Emil-18
